src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -165,12 +194,54 @@ private Issue createJiraIssue(AbstractBuild<?, ?> build, String filename) throws
         );
         Iterable<String> components = Splitter.on(",").trimResults().omitEmptyStrings().split(component);
 
-        Issue issue = session.createIssue(projectKey, description, assignee, components, summary);
+        Long issueTypeId = getIssueTypeId(session, type);
+        Long priorityId = getPriorityId(session, priority);
+
+        Issue issue = session.createIssue(projectKey, description, assignee, components, summary, issueTypeId, priorityId);
 
         writeInFile(filename, issue);
         return issue;
     }
 
+    @Nonnull
+    private Long getIssueTypeId(JiraSession session, String type) {
+        if (StringUtils.isNotBlank(type)) {
+            for (IssueType t : session.getIssueTypes()) {
+                if (type.equalsIgnoreCase(t.getName())) {
+                    return t.getId();
+                }
+            }
+            if (isNumber(type)) {
+                return Long.valueOf(type.trim());
+            }
+        }
+        return JiraRestService.BUG_ISSUE_TYPE_ID;
+    }
+
+    @Nullable
+    private Long getPriorityId(JiraSession session, String priority) {
+        if (StringUtils.isNotBlank(priority)) {
+            for (Priority p : session.getPriorities()) {
+                if (priority.equalsIgnoreCase(p.getName())) {
+                    return p.getId();
+                }
+            }
+            if (isNumber(priority)) {
+                return Long.valueOf(priority.trim());
+            }
+        }
+        return null;
+    }
+
+    private final Pattern pattern = Pattern.compile("-?\\d+");
```

Comment:
to the top and more meaningful variable name :-)
(and maybe the actual regex in a constant?)